src/data/sources/steam/Steam.vala

```diff
@@ -495,56 +507,267 @@ namespace GameHub.Data.Sources.Steam
+
+			if(game.image != null)
+			{
+				try
+				{
+					var cached = ImageCache.local_file(game.image, @"games/$(game.source.id)/$(game.id)/images/");
+					var dest = FSUtils.file(get_userdata_dir().get_child("config").get_child("grid").get_path(), custom_appid + ".png");
+					cached.copy(dest, FileCopyFlags.OVERWRITE);
+				}
+				catch (Error e) {}
+			}
+
 			if(game.image_vertical != null)
 			{
 				try
 				{
 					var cached = ImageCache.local_file(game.image_vertical, @"games/$(game.source.id)/$(game.id)/images/");
-					// https://github.com/boppreh/steamgrid/blob/master/games.go#L120
-					uint64 id = crc32(0, (ProjectConfig.PROJECT_NAME + game.name).data) | 0x80000000;
-					var dest = FSUtils.file(get_userdata_dir().get_child("config").get_child("grid").get_path(), id.to_string() + "p.png");
-					cached.copy(dest, NONE);
+					var dest = FSUtils.file(get_userdata_dir().get_child("config").get_child("grid").get_path(), custom_appid + "p.png");
+					cached.copy(dest, FileCopyFlags.OVERWRITE);
 				}
 				catch (Error e) {}
 			}
+		}
 
-			var tags_node = new BinaryVDF.ListNode.node("tags");
-			tags_node.add_node(new BinaryVDF.StringNode.node("0", "GameHub"));
+		// add or update collections
+		private static void set_shortcut_collection(Game game)
+		{
+			string? error;
+
+			var config_dir = FSUtils.find_case_insensitive(get_userdata_dir(), "config");
+			if(config_dir == null || !config_dir.query_exists()) return;
+
+			var collections_db = new SteamCollectionDatabase(instance.user_id, out error);
+			if(error != null) return;
+
+			collections_db.read(out error);
+			if(error != null) return;
+
+			var localconfig = Parser.parse_vdf_file(config_dir.get_path(), "localconfig.vdf");
+			if(localconfig == null || localconfig.get_node_type() != Json.NodeType.OBJECT) return;
+			var user_collections = Parser.parse_json(localconfig.get_object().get_object_member("UserLocalConfigStore").get_object_member("WebStorage").get_string_member("user-collections"));
+			if(user_collections == null || user_collections.get_node_type() != Json.NodeType.OBJECT) return;
+
+			int64? custom_appid_int;
+			try
+			{
+				// https://github.com/boppreh/steamgrid/blob/master/games.go#L120
+				var custom_appid = (crc32(0, (ProjectConfig.PROJECT_NAME + game.name).data) | 0x80000000).to_string();
+				int64.from_string(custom_appid, out custom_appid_int);
+				if(custom_appid_int == null) return;
+			}
+			catch (Error e) {return;}
+
+			// Remove from collections where the game doesn't have the tag anymore
+			user_collections.get_object().foreach_member((object, name, node) =>
+			{
+				if(name.has_prefix("gh-") || name == "favorite" || name == "hidden")
+				{
+					foreach(var tag in game.tags)
+					{
+						string key;
+						if(tag.id == "builtin:favorites")
+						{
+							key = "favorite";
+						}
+						else if(tag.id == "builtin:hidden")
+						{
+							key = "hidden";
+						}
+						else
+						{
+							key = @"gh-$(tag.name)";
```

Comment:
Should `tag.id` be used here? `id` is `"user:" + md5(tag.name)` for user-defined tags.

Reply:
Probably better. Tested with special characters from gamehub to steam and had no problems but since you've mentioned this I tested it the other way around and it seems there's a problem parsing into json. I'll have a look into it.